### PR TITLE
Make `no_std` compatible

### PR DESCRIPTION
**Pull request by MechSlayer**
- Made `no_std` compatible by directly using `write!` in the `fmt` impl
- Add support for generics, lifetimes and where clauses

### CODE REVIEW

**Comment by Rafaeltheraven** [comment]
Thank you very much for the PR and fixing my oversights on these fronts. `logos_display` was one of my first forays into Rust and procedural macros and you can really tell at points...

File: src/lib.rs [hunk 1]
```diff
@@ -58,51 +57,64 @@ fn _logos_display(input: TokenStream2, debug: bool) -> TokenStream2 {
...
                 }
             }
-        },
+        }
         Err(e) => e.to_compile_error()
     }
 }
 
-fn gen_anon_args(n: usize) -> TokenStream2 {
+fn gen_anon_args(n: usize) -> Vec<proc_macro2::TokenStream> {
```
Comment (Rafaeltheraven) [inline src/lib.rs]: Small nitpick. Could you change this type to `Vec<TokenStream2>` so that it is consistent with the rest of the crate?

Comment (MechSlayer) [reply src/lib.rs]: Sure thing, I'll also run `cargo fmt` since I forgot earlier.

**Comment by Rafaeltheraven** [comment]
Looks good to me! Will update the README accordingly and publish to crates.io.

**Pull request merged**